### PR TITLE
AGPUSH-1250 : several polishing

### PR DESCRIPTION
**Pull request by sebastienblanc**
@matzew Could you review ?

### CODE REVIEW

File: src/main/java/org/jboss/aerogear/adm/internal/Utilities.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import com.google.gson.JsonObject;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.Charset;
+
+public class Utilities {
```
Comment (matzew) [inline src/main/java/org/jboss/aerogear/adm/internal/Utilities.java]: let's make it final and a private ctor

**Comment by matzew** [comment]
Not sure I get this part:
https://github.com/aerogear/java-adm/blob/AGPUSH-1250/src/main/java/org/jboss/aerogear/adm/AdmService.java#L77-L89

we are not doing anything with the read fields ?

**Comment by matzew** [comment]
move https://github.com/aerogear/java-adm/blob/AGPUSH-1250/src/main/java/org/jboss/aerogear/adm/TokenService.java#L43 to line above 

https://github.com/aerogear/java-adm/blob/AGPUSH-1250/src/main/java/org/jboss/aerogear/adm/AdmService.java#L113 formating

**Comment by sebastienblanc** [comment]
@matzew Regarding https://github.com/aerogear/java-adm/blob/AGPUSH-1250/src/main/java/org/jboss/aerogear/adm/AdmService.java#L77-L89 , indeed we are not doing a lot of that. Looking at ADM specs I see 

```
registrationID: The current registration ID of the app instance. If this value is different than the one passed in by your server, your server must update its records to use this value.

```

Maybe we can change the signature of the method to return this registrationID. And in UPS, for instance, if the value has changed we can update the installation.

**Comment by matzew** [comment]
> Maybe we can change the signature of the method to return this registrationID. And in UPS, for instance, if the value has changed we can update the installation.

That sounds more reasonable to me.

File: src/main/java/org/jboss/aerogear/adm/TokenService.java [hunk 2]
```diff
@@ -47,44 +39,58 @@
...
+        StringBuilder builder = new StringBuilder();
+        builder
+                .append("grant_type=")
+                .append(URLEncoder.encode(Utilities.CLIENT_CREDENTIALS, Utilities.UTF_8) )
+                .append("&")
+                .append("scope=")
+                .append( URLEncoder.encode(Utilities.MESSAGING_PUSH, Utilities.UTF_8) )
+                .append("&")
+                .append("client_id=")
+                .append("URLEncoder.encode(clientId, Utilities.UTF_8)")
```
Comment (matzew) [inline src/main/java/org/jboss/aerogear/adm/TokenService.java]: @sebastienblanc is this string `"URLEncoder.encode(clientId, Utilities.UTF_8)"` really wanted ? :-)

**Pull request merged**